.github/workflows/ci.yml

```diff
@@ -153,7 +153,82 @@ jobs:
           sed -i "s/image: flink:.*/image: ${{ matrix.versions.image }}/" e2e-tests/data/*.yaml
           sed -i "s/flinkVersion: .*/flinkVersion: ${{ matrix.versions.flinkVersion }}/" e2e-tests/data/*.yaml
           git diff HEAD
-          ls e2e-tests/test_*.sh | while read script_test;do \
+          ls e2e-tests/test_application_*.sh e2e-tests/test_sessionjob_*.sh | while read script_test;do \
+            echo "Running $script_test"
+            bash $script_test || exit 1
+          done
+          git reset --hard
+      - name: Stop the operator
+        run: |
+          helm uninstall -n ${{ matrix.config.namespace }} flink-kubernetes-operator
+      - name: Stop minikube
+        run: |
+          source e2e-tests/utils.sh
+          stop_minikube
+  e2e_multi_ci:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        config:
+          - description: 'WatchNamespaces enabled'
+            namespace: flink
+            extraArgs: '--create-namespace --set "watchNamespaces={default,flink}"'
+        versions:
+          - image: flink:1.13
+            flinkVersion: v1_13
+          - image: flink:1.14
+            flinkVersion: v1_14
+          - image: flink:1.15
+            flinkVersion: v1_15
+    name: e2e_multi_ci
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+      - name: Cache local Maven repository
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - name: Build with Maven
+        run: |
+          set -o pipefail; mvn clean install -DskipTests | tee ./mvn.log; set +o pipefail
+          if [[ $(cat ./mvn.log | grep -E -v 'flink-runtime-.*.jar, flink-kubernetes-operator-.*.jar define 2 overlapping classes' | grep -c "overlapping classes" -) -gt 0 ]];then
+            echo "Found overlapping classes, please fix"
+            exit 1
+          fi
+      - name: Start minikube
+        run: |
+          source e2e-tests/utils.sh
+          start_minikube
+      - name: Install cert-manager
+        run: |
+          kubectl get pods -A
+          kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.7.1/cert-manager.yaml
+          kubectl -n cert-manager wait --all=true --for=condition=Available --timeout=300s deploy
+      - name: Build image
+        run: |
+          export SHELL=/bin/bash
+          export DOCKER_BUILDKIT=1
+          eval $(minikube -p minikube docker-env)
+          docker build --progress=plain --no-cache -f ./Dockerfile -t flink-kubernetes-operator:ci-latest --progress plain .
```

Comment:
nit: Is there some way to avoid building twice image

Reply:
I'll try to simply add the tests to the existing e2e_ci workflow, we can probably optimize the tests to run parallel without building the images, I'll open a follow up if I succeed